### PR TITLE
Cleanup locationcontroller

### PR DESCRIPTION
**Pull request by Henni**
This pull-request removes the deprecated function `$this->params()` from the locationcontroller and cleans up few other things.

### CODE REVIEW

**Comment by jancborchardt** [comment]
Best for @DJaeger and @brantje to review :)

**Comment by jancborchardt** [comment]
@Henni can you rebase this?

@DJaeger @brantje @v1r0x @vgezer @povoq can you review? :)

**Comment by DJaeger** [comment]
Where do you got the information from, that ```$this->params()``` is depricated?

As of the docu is to be used:
http://api.owncloud.org/classes/OCP.AppFramework.ApiController.html#params

How should the method calls otherwise determine in which order the params have to be applied?

**Comment by DJaeger** [comment]
The removal of TemplateResponse and addtion of LocationManager is ok.

**Comment by Henni** [comment]
@DJaeger https://github.com/owncloud/core/blob/master/lib/public/appframework/controller.php#L157
I implemented it like this in the tasks app as well.

**Comment by LukasReschke** [comment]
> How should the method calls otherwise determine in which order the params have to be applied?

That's some magic we do with reflection in core. Let me point you to the source code in core, always good to understand how and why things work like they do :smile: 

In https://github.com/owncloud/core/blob/c700f42b68f430e9c89ce97b92ea91323c9f6ed5/lib/private/appframework/http/dispatcher.php we do read the parameters there and do call the function with the right parameters then :)

File: controller/locationcontroller.php [hunk 1]
```diff
@@ -40,23 +39,22 @@ public function __construct($appName, IRequest $request,$locationManager,$userId
...
-	public function update() {
-		$params = array('user' => $this -> userId);
-		$location['lat'] = $this->params('lat');
-		$location['lng'] = $this->params('lon');
-		if(((string)(float)$this->params('timestamp') === $this->params('timestamp'))) {
-			if(strtotime(date('d-m-Y H:i:s',$this->params('timestamp'))) === (int)$this->params('timestamp')) {
-				$location['timestamp'] = (int)$this->params('timestamp');
-			} elseif(strtotime(date('d-m-Y H:i:s',$this->params('timestamp')/1000)) === (int)floor($this->params('timestamp')/1000)) {
-				$location['timestamp'] = (int)floor($this->params('timestamp')/1000);
+	public function update($lat, $lon, $timestamp, $hdop, $altitude, $speed, $hash) {
```
Comment (LukasReschke) [inline controller/locationcontroller.php]: Update PHPDoc?

Also why are there still the insecure defaults? :speak_no_evil:

File: controller/locationcontroller.php [hunk 2]
```diff
@@ -86,11 +84,11 @@ public function loadDevices(){
 	 * @NoAdminRequired
 	 * @NoCSRFRequired
 	 */		
-	public function loadLocations(){
-		$deviceIds = explode(',',$this->params('devices'));
-		$from = ($this->params('from')) ? strtotime($this->params('from')) : null;
-		$till = ($this->params('till')!='') ? strtotime($this->params('till')) : strtotime('now');
-		$limit = ($this->params('limit')!='') ? (int) $this->params('limit') : 2000;
+	public function loadLocations($devices, $from, $till, $limit){
```
Comment (LukasReschke) [inline controller/locationcontroller.php]: Update PHPDoc?

File: controller/locationcontroller.php (hunk 1 above)
Comment (Henni) [inline controller/locationcontroller.php]: @LukasReschke By insecure defaults you mean:
 * @&#8203;NoAdminRequired
 * @&#8203;NoCSRFRequired
 * @&#8203;CORS
 * @&#8203;PublicPage
?

Comment (LukasReschke) [inline controller/locationcontroller.php]: Yeah. You really do at least want to remove the NoCSRFRequired… Also if an user i srquired drop the PublicPage.

CSRF = any website can send requests to others using JS for example, cookies are included. Thus we need to always send a random token that OC verifies. Our JS does this automatically for you :-)

Btw. if you're coming to the conf we can have an in-depth discussion about such things if you want :-)

Comment (Henni) [inline controller/locationcontroller.php]: @LukasReschke I'd really like to hear about some of this cross site stuff from you.

Comment (LukasReschke) [inline controller/locationcontroller.php]: https://conference.owncloud.org/conference/oCC2015/proposal/59 :-)

(and please bring your laptop with you ;))

Comment (LukasReschke) [inline controller/locationcontroller.php]: Also @DJaeger, you are more than welcome if you're interested :-)

(And obviously everybody else, but them I can't mention and force to come)

Comment (Henni) [inline controller/locationcontroller.php]: @LukasReschke Awesome! Looking forward to it

**Comment by Henni** [comment]
I removed all the insecure defaults and added simple PHPDocs.

**Comment by DJaeger** [comment]
Now looks great. :+1:
And thanks for the clarification and invitation, @LukasReschke. :-)

**Comment by LukasReschke** [comment]
YOLO

**Pull request merged**